app/uk/gov/hmrc/agentservicesaccount/auth/PasscodeVerification.scala

```diff
@@ -47,36 +46,33 @@ class FrontendPasscodeVerification @Inject()(configuration: Configuration,
   lazy val verificationURL: String = configuration.getString(s"govuk-tax.$env.url.verification-frontend.redirect").getOrElse("/verification")
   lazy val logoutUrl = s"$verificationURL/otac/logout/$passcodeRegime"
 
-  def loginUrl[A](queryParam: String) = s"$verificationURL/otac/login$queryParam"
+  private def redirectToLoginWithToken[A](implicit request: Request[A], ec: ExecutionContext): Option[Future[Result]] = {
```

Comment:
Its not necessary to make the return type Future at this moment. You can enclose the result of this method in a future when its been used.